x-pack/legacy/plugins/uptime/public/state/api/monitor_status.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+// import { ThrowReporter } from 'io-ts/lib/ThrowReporter';
+// import { isRight } from 'fp-ts/lib/Either';
+import { getApiPath } from '../../lib/helper';
+// import { SnapshotType, Snapshot } from '../../../common/runtime_types';
+import { QueryParams } from '../actions/types';
+import { Ping } from '../../../common/graphql/types';
+
+export interface APIParams {
+  basePath: string;
+  monitorId: string;
+}
+
+export const fetchMonitorStatus = async ({
+  basePath,
+  monitorId,
+  dateStart,
+  dateEnd,
+  location,
+}: QueryParams & APIParams): Promise<Ping> => {
+  const url = getApiPath(`/api/uptime/monitor/status`, basePath);
+  const params = {
+    monitorId,
+    dateStart,
+    dateEnd,
+    ...(location && { location }),
+  };
+  const urlParams = new URLSearchParams(params).toString();
+  const response = await fetch(`${url}?${urlParams}`);
```

Comment:
Maybe we should add a TODO here to use core HTTP services when https://github.com/elastic/kibana/pull/54076/files is merged.

Reply:
We should discuss the best way for the API handlers to consume the Kibana core http service. We could accept the service as a parameter of these functions, and access it via the `useKibana` hook in the caller code, or dispatch an action with a reference to the service from a top-level effect, or implement some other strategy. In any case, it'll enable us to delete a bunch of code around ensuring we use the `basePath` for our API calls.